pkg/controller/datavolume-controller_test.go

```diff
@@ -1581,6 +1581,179 @@ var _ = Describe("All DataVolume Tests", func() {
+			dv := newCloneDataVolumeWithEmptyStorage("test-dv", "default")
+			cloneStrategy := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyHostAssisted)
+			storageProfile := createStorageProfileWithCloneStrategy(scName, []cdiv1.ClaimPropertySet{
+				{AccessModes: accessMode, VolumeMode: &blockMode}}, &cloneStrategy)
+
+			pvc := createPvcInStorageClass("test", metav1.NamespaceDefault, &scName, nil, nil, corev1.ClaimBound)
+			pvc.SetAnnotations(make(map[string]string))
+			pvc.GetAnnotations()[AnnPodPhase] = string(corev1.PodSucceeded)
+			reconciler := createDatavolumeReconciler(dv, pvc, storageProfile, sc)
+
+			// Prepare the size-detection Pod with the required information
+			pod := reconciler.makeSizeDetectionPodSpec(pvc, dv)
+			pod.Status.Phase = corev1.PodSucceeded
+			err := reconciler.client.Create(context.TODO(), pod)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Checks
+			pvcSpec, err := RenderPvcSpec(reconciler.client, reconciler.recorder, reconciler.log, dv)
+			Expect(err).ToNot(HaveOccurred())
+			err = reconciler.detectCloneSize(dv, pvcSpec, HostAssistedClone)
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(Equal(ErrNonValidTermMsg))
+		})
+
+		It("Should get the size from the size-detection pod", func() {
+			dv := newCloneDataVolumeWithEmptyStorage("test-dv", "default")
+			cloneStrategy := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyHostAssisted)
+			storageProfile := createStorageProfileWithCloneStrategy(scName, []cdiv1.ClaimPropertySet{
+				{AccessModes: accessMode, VolumeMode: &blockMode}}, &cloneStrategy)
+
+			pvc := createPvcInStorageClass("test", metav1.NamespaceDefault, &scName, nil, nil, corev1.ClaimBound)
+			pvc.SetAnnotations(make(map[string]string))
+			pvc.GetAnnotations()[AnnPodPhase] = string(corev1.PodSucceeded)
+			reconciler := createDatavolumeReconciler(dv, pvc, storageProfile, sc)
+
+			// Prepare the size-detection Pod with the required information
+			pod := reconciler.makeSizeDetectionPodSpec(pvc, dv)
+			pod.Status.Phase = corev1.PodSucceeded
+			pod.Status.ContainerStatuses = []corev1.ContainerStatus{
+				{
+					State: corev1.ContainerState{
+						Terminated: &corev1.ContainerStateTerminated{
+							ExitCode: 0,
+							Message:  "100", // Mock value
+						},
+					},
+				},
+			}
+			err := reconciler.client.Create(context.TODO(), pod)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Get the expected value
+			pvcSpec, err := RenderPvcSpec(reconciler.client, reconciler.recorder, reconciler.log, dv)
+			Expect(err).ToNot(HaveOccurred())
+			expectedSize, err := inflateSizeWithOverhead(reconciler.client, int64(100), pvcSpec)
+
+			// Checks
+			err = reconciler.detectCloneSize(dv, pvcSpec, HostAssistedClone)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(dv.GetAnnotations()[AnnPermissiveClone]).To(Equal("true"))
+			targetSize := pvcSpec.Resources.Requests[corev1.ResourceStorage]
+			Expect(targetSize.Cmp(expectedSize)).To(Equal(0))
```

Comment:
This works, alternative is:
```go
Expect(targetSize.Int64()).To(Equal(int64(100))
```
Not 100% sure if Int64 is a single or double value return function.

Reply:
It is probably 5th time you need to Rebase, so I think you can leave it as is, because the PR gets bigger and bigger and now it would be good to focus only on important stuff and try to finish it and merge it. 

And keep the details noted somewhere maybe on a new issue, and then we can open follow up PR with some small cleanups. You decide. I prefer to have things fixed the first time. But I also like to have stuff done, and not blocked by such small details.